src/show/components/TextField/index.js

```diff
@@ -1,23 +1,79 @@
-import React from "react";
-
+import React, { useState } from 'react';
 const TextField = ({
+  name,
+  type = 'text',
   label = "",
   placeholder = "",
-  required = false,
   value = "",
-  password = false,
-}) => (
-  <div>
-    <h6>{label}</h6>
-    <input
-      placeholder={placeholder}
-      type={password ? "password" : "text"}
-      password={true}
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
+  onChange,
+  hasEye,
+}) => {
+  const [error, setError] = useState('');
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (value) => {
+    if (type === 'email') {
+      const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+      if (!emailPattern.test(value)) {
+        setError('Invalid email address');
+        return;
+      }
+    } else if (type === 'password') {
+      const passwordPattern = /^(?=.*[0-9])(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,}$/;
+      if (!passwordPattern.test(value)) {
+        setError('Password must be at least 8 characters long and include at least one number and one special character');
+        return;
+      }
+    }
+    setError('');
+  };
+
+  const handleChange = (e) => {
+    const { value } = e.target;
+    validate(value);
+    onChange(value);
+  };
+
+  const togglePasswordVisibility = () => {
+    setInputType(inputType === 'password' ? 'text' : 'password');
+  };
+
+  return (
+    <div className="relative mb-4">
+      {label && (
+        <label className="block mb-1 font-medium text-zinc-400" style={{ fontSize: "0.6rem" }}>
+          {label}
+        </label>
+      )}
+      <input
+        type={inputType}
+        name={name}
+        placeholder={placeholder}
+        value={value}
+        onChange={handleChange}
+        className={`w-full border rounded-md text-xs p-1 ${error ? 'border-red-500' : 'border-gray-300'} focus:outline-none`}
+      />
+      {hasEye && type === 'password' && (
+        <button
+          type="button"
+          onClick={togglePasswordVisibility}
+          className="absolute pt-2 right-2"
+        >
+          {inputType === 'password' ? (
+            <svg className="h-2.5 w-2.5 text-gray-600" fill="none" stroke="currentColor" viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
```

Comment:
use svg as component and save all this in assets under svg folder and create an index.js file there

Reply:
done